pandas/core/array_algos/take.py

```diff
@@ -130,7 +135,8 @@ def take_1d(
     """
     Specialized version for 1D arrays. Differences compared to take_nd:
 
-    - Assumes input (arr, indexer) has already been converted to numpy array / EA
+    - Assumes input array has already been converted to numpy array / EA
+    - Assumes indexer is already guaranteed to be int64 dtype ndarray
     - Only works for 1D arrays
 
```

Comment:
entry for 'validate_indices has already been checked'?

Reply:
> entry for 'validate_indices has already been checked'?

Added (that's the same for `take_nd` as well though, so it's not a difference with take_nd)

Reply:
thanks.  I _think_ calling take_1d without having done this check was the source of the segfaults i was getting